README.md

```diff
@@ -136,6 +136,14 @@ $ TEST_INTEGRATION=1 TEST_DELETE_RECORDS=1 KAFKA_CONNECT=localhost:9094 cargo te
 in another session. Note that Apache Kafka supports a different set of features then redpanda, so we pass other
 environment variables.
 
+### Using a SOCKS 5 Proxy
+
+To run the integration test via a SOCKS 5 proxy, you need to set the environment variable `SOCKS_PROXY`. The following command requires a running proxy on the local machine.
+```console
+$ KAFKA_CONNECT=0.0.0.0:9093 SOCKS_PROXY=localhost:8080 cargo test --features full
+```
+To avoid known issues with the [test library](https://github.com/yanganto/test-with/issues/18), it is recommended to clean up before the first test.
```

Comment:
Sure, I changed that locally to use a macro. However the drawback of this approach is that you don't see that the test is skipped.

Reply:
Yeah, the macro is not ideal. My hope is that https://github.com/rust-lang/rfcs/pull/3221 will be accepted and we can get the best of both worlds. But until then the skipping won't be reported properly.

Reply:
Hi, this is something you may want to use for this scenario.
https://crates.io/crates/test-with
If you are using CI tool for testing, or under development with source code changing, you do not need cargo clean before testing. 

If there is any issue on the crate, an issue is always welcome.  :)

Reply:
By the way, there is more than one issue in `#[ignore(if)]`, so it might not be the better solution to solve.  It is good to keep ignoring in runtime and ignoring under conditions separately.

The following are related PR and issues in Rust, for you to know more about these if you are interested.
- https://github.com/rust-lang/rust/issues/68007
- https://github.com/rust-lang/rust/pull/96574

Reply:
I'm not to worried about the CI and the env variable handling there. Ignoring tests and build time instead of runtime is IMHO a messy DX (developer experience). Rskafka is already a somewhat complicated crate -- especially when you consider the integration tests. Sure, not having a proper "ignore" status for skipped tests is bad DX as well, but I consider this the lesser evil.

So let's wait until Rust implements the required feature. I've created #139 to track this on our side.

Reply:
You could assign the #139 to me as well, I would be more than happy to integrate the feature, when Rust is ready.